syntax = "proto3";

message DataPack{
  string type = 1;
  string id = 2;
  double timeStamp = 3;
  LoginData login_data = 4;
  LogoutData logout_data = 5;
  AckData ack_data = 6;
  UserInfo user_info = 7;
  Msg one_msg = 8;
  repeated UserInfo friends_info = 9;
  repeated GroupInfo groups_info = 10;
  repeated Msg offline_msg = 11;
}
message LoginData{
  string uid = 1;
  string password = 2;
}
message LogoutData{
  string uid = 1;
  string session = 2;
}
message AckData{
  string ack_id = 1;
  bool login_status = 2;
  string session = 3;
}
message UserInfo{
  string uid = 1;
  string nick_name = 2;
  bool  isMale = 3;
  double registration_time = 4;
  double start_time = 5;
  bytes avatar = 6;
}
message GroupInfo{
  string g_id = 1;
  string g_name = 2;
  double creat_time = 3;
  string g_author_id = 4;
  string g_nick_name = 5;
}
message Msg{
  string from_id = 1;
  string to_id = 2;
  string msg_type = 3;
  string text = 4;
  bytes image = 5;
}